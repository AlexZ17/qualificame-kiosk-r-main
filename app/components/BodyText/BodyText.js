import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './BodyTextStyle';
import { Fonts, Colors } from 'qualificame-kiosk/app/styles';

export default class BodyText extends PureComponent {
	render() {
		const textColor = { color: Colors[this.props.color] };
		const weight = { fontWeight: Fonts.weight[this.props.weight] };
		const align = { textAlign: this.props.align };
		
		return (
			<View>
				<Text style={[styles.bodyText, textColor, weight, align]}>
					{this.props.text}
				</Text>
			</View>
		);
	}
}

	BodyText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	BodyText.defaultProps = {
		text: '',
		color: "black",
		weight: "regular",
		align: "left"
	}
