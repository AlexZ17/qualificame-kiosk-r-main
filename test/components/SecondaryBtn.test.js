import React from 'react';
import { shallow } from 'enzyme';
import { TouchableOpacity } from 'react-native';

import BodyText from 'qualificame-kiosk/app/components/BodyText';
import SecondaryBtn from 'qualificame-kiosk/app/components/SecondaryBtn';

function createTestProps (props) {
	return {
		text: 'Hello World',
		disabled: true,
		color: 'blue',
		onPress: ()=>{},
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<SecondaryBtn {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('renders the test text inside', () => {
		expect(wrapper.find(BodyText).prop('text') ).toEqual('Hello World')
	});
	it('Text have the correct color', () => {
		expect(wrapper.find(BodyText).prop('color') ).toEqual('blue');
	});
	it('if touchableOpacity has disabled', () => {
		expect(wrapper.find(TouchableOpacity).prop('disabled') ).toBeTruthy();
	});
});

describe('interactions', () => {
	let wrapper;
	const createWrapper = props => shallow(<SecondaryBtn {...props}/>);
	describe('when onPress function is passed to props', () => {
		let fn;
		beforeEach(() => {
			fn = jest.fn();
			const props = createTestProps({ onPress: fn })
			wrapper = createWrapper(props);
		});
		it('it is called on button press', () => {
			wrapper.find(TouchableOpacity).first().props().onPress();
			expect(fn).toBeCalled();
		});
	});
})