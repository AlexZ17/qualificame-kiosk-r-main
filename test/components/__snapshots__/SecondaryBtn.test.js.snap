// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  disabled={true}
  onPress={[Function]}
>
  <Component
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 10,
          "borderWidth": 1,
          "padding": 8,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "borderColor": "#146DCA",
        },
      ]
    }
  >
    <BodyText
      align=""
      color="blue"
      text="Hello World"
      weight="medium"
    />
  </Component>
</TouchableOpacity>
`;
